### DISCUSSION TITLE
Getting parsing error while providing bitbucket cloud config

### DISCUSSION THREAD

**Discussion by ssharma2089** [renovatebot/renovate#11959] [Request Help]
### How are you running Renovate?

Self-hosted

### Please select which platform you are using if self-hosting.

Bitbucket Cloud (bitbucket.org)

### If you're self-hosting Renovate, tell us what version of Renovate you run.

27.14.2

### Describe the bug

I have deployed renovate through helm charts and created config.json to provide bitbucket configuration. Please find the below config.json

module.exports = {
  platform: 'bitbucket',
  username: 'user1',
  password:'*********',
  onboarding: true,
  gitAuthor: "Renovate Bot <bot@renovateapp.com>",
  onboardingConfig: {
    extends: ['config:base'],
  },
  repositories: [ 'user/reno' ],
}

Could you please confirm Is there anything wrong while passing configuration

### Relevant debug logs

```Could not parse config file 
        SyntaxError: /usr/src/app/config.json: Unexpected token m in JSON at position 1
           at parse (<anonymous>)
           at Object.Module._extensions..json (internal/modules/cjs/loader.js:1115:22)
           at Module.load (internal/modules/cjs/loader.js:937:32)
           at Function.Module._load (internal/modules/cjs/loader.js:778:12)
           at Module.require (internal/modules/cjs/loader.js:961:19)
           at require (internal/modules/cjs/helpers.js:92:18)
           at Object.getConfig (/usr/src/app/node_modules/renovate/lib/workers/global/config/parse/file.ts:15:14)
           at parseConfigs (/usr/src/app/node_modules/renovate/lib/workers/global/config/parse/index.ts:19:33)
           at getGlobalConfig (/usr/src/app/node_modules/renovate/lib/workers/global/index.ts:42:22)
           at Object.start (/usr/src/app/node_modules/renovate/lib/workers/global/index.ts:87:20)
           at /usr/src/app/node_modules/renovate/lib/renovate.ts:16:41
           at Object.<anonymous> (/usr/src/app/node_modules/renovate/lib/renovate.ts:21:3)
           at Module._compile (internal/modules/cjs/loader.js:1072:14)
           at Object.Module._extensions..js (internal/modules/cjs/loader.js:1101:10)
           at Module.load (internal/modules/cjs/loader.js:937:32)
           at Function.Module._load (internal/modules/cjs/loader.js:778:12)
           at Module.require (internal/modules/cjs/loader.js:961:19)
           at require (internal/modules/cjs/helpers.js:92:18)
           at Object.<anonymous> (/usr/src/app/src/renovate.ts:9:1)
           at Module._compile (internal/modules/cjs/loader.js:1072:14)



### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by viceice** [comment]
1. Please format your code
2. you can't add JavaScript code as JSON, so your code should be added as  `config.js` or convert it to JSON.

**Reply by ssharma2089** [reply]
@viceice  After adding config.js. Getting below error

      "durationMs": 27089
 INFO: Renovate is exiting with a non-zero code due to the following logged errors
       "loggerErrors": [
         {
           "name": "renovate",
           "level": 50,
           "logContext": "uKc9oa1XK",
           "repository": "sagar2089/reno",
           "err": {
             "name": "RequestError",
             "code": "EAI_AGAIN",
             "timings": {
               "start": 1633002339260,
               "socket": 1633002339260,
               "lookup": 1633002347272,
               "error": 1633002347272,
               "phases": {"wait": 0, "dns": 8012, "total": 8012}
             },
             "message": "getaddrinfo EAI_AGAIN api.bitbucket.org",
             "stack": "RequestError: getaddrinfo EAI_AGAIN api.bitbucket.org\n    at ClientRequest.<anonymous> (/usr/src/app/node_modules/got/dist/source/core/index.js:956:111)\n    at Object.onceWrapper (events.js:520:26)\n    at ClientRequest.emit (events.js:412:35)\n    at ClientRequest.emit (domain.js:470:12)\n    at ClientRequest.origin.emit (/usr/src/app/node_modules/@szmarczak/http-timer/dist/source/index.js:43:20)\n    at TLSSocket.socketErrorListener (_http_client.js:475:9)\n    at TLSSocket.emit (events.js:400:28)\n    at TLSSocket.emit (domain.js:470:12)\n    at emitErrorNT (internal/streams/destroy.js:106:8)\n    at emitErrorCloseNT (internal/streams/destroy.js:74:3)\n    at processTicksAndRejections (internal/process/task_queues.js:82:21)\n    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:69:26)",
             "options": {
               "headers": {
                 "user-agent": "RenovateBot/27.14.2 (https://github.com/renovatebot/renovate)",
                 "authorization": "***********",
                 "accept": "application/json",
                 "accept-encoding": "gzip, deflate, br"
               },
               "url": "https://api.bitbucket.org/2.0/repositories/sagar2089/reno",
               "hostType": "bitbucket",
               "username": "",
               "password": "",
               "method": "GET",
               "http2": false
             }
           },
           "msg": "Repository has unknown error"

**Reply by viceice** [reply]
Check your system settings, there is a DNS failure

**Reply by ssharma2089** [reply]
@viceice Yes, you are right. Getting below error
 DEBUG: Failed to retrieve default.json from repo sagar2089/renovate-config (repository=sagar2089/reno)
       "err": {
         "name": "HTTPError",
         "timings": {
           "start": 1633086250903,
           "socket": 1633086250904,
           "lookup": 1633086250906,
           "connect": 1633086251132,
           "secureConnect": 1633086251582,
           "upload": 1633086251582,
           "response": 1633086251882,
           "end": 1633086251883,
           "phases": {
             "wait": 1,
             "dns": 2,
             "tcp": 226,
             "tls": 450,
             "request": 0,
             "firstByte": 300,
             "download": 1,
             "total": 980
           }
         },
         "message": "Response code 404 (NOT FOUND)",
         "stack": "HTTPError: Response code 404 (NOT FOUND)\n    at Request.<anonymous> (/usr/src/app/node_modules/got/dist/source/as-promise/index.js:117:42)\n    at processTicksAndRejections (internal/process/task_queues.js:95:5)",
         "options": {
           "headers": {
             "user-agent": "RenovateBot/25.69.4 (https://github.com/renovatebot/renovate)",
             "authorization": "***********",
             "accept-encoding": "gzip, deflate, br"
           },
           "url": "https://api.bitbucket.org/2.0/repositories/sagar2089/renovate-config/src/HEAD/default.json",
           "username": "",
           "password": "",
           "method": "GET",
           "http2": false
         },
         "response": {
           "statusCode": 404,
           "statusMessage": "NOT FOUND",
           "body": "{\"type\":\"error\",\"error\":{\"message\":\"The requested repository either does not exist or you do not have access. If you believe this repository exists and you have access, make sure you're authenticated.\"}}",
           "headers": {
             "server": "nginx",
             "vary": "Authorization, Origin, cookie, user-context",
             "cache-control": "private",
             "content-type": "application/json; charset=utf-8",
             "x-b3-traceid": "c50eb1497929b0e7",
             "x-oauth-scopes": "runner:write, pipeline:variable, webhook, snippet:write, wiki, issue:write, pullrequest:write, repository:delete, repository:admin, project:write, team:write, account:write",
             "x-usage-output-ops": "0",
             "x-dc-location": "Micros",
             "strict-transport-security": "max-age=31536000; includeSubDomains; preload",
             "date": "Fri, 01 Oct 2021 11:04:11 GMT",
             "x-usage-user-time": "0.034189",
             "x-usage-system-time": "0.000000",
             "x-served-by": "8fbec6289d11",
             "x-view-name": "bitbucket.apps.repo2.api.v20.src.SourceHandler",
             "etag": "\"750cee10d61dc8c86530b796fe293fb1\"",
             "x-static-version": "b77f2172789c",
             "x-credential-type": "apppassword",
             "x-render-time": "0.0599088668823",
             "x-accepted-oauth-scopes": "repository",
             "connection": "close",
             "x-usage-input-ops": "0",
             "x-version": "b77f2172789c",
             "x-request-count": "3607",
             "x-frame-options": "SAMEORIGIN",
             "content-length": "203"
           },
           "httpVersion": "1.1"
         }
       },
       "repo": "sagar2089/renovate-config",
       "fileName": "default.json"
 INFO: Fallback to renovate.json file as a preset is deprecated, please use a default.json file instead. (repository=sagar2089/reno)
DEBUG: Failed to retrieve renovate.json from repo sagar2089/renovate-config (repository=sagar2089/reno)
       "err": {
         "name": "HTTPError",
         "timings": {
           "start": 1633086251896,
           "socket": 1633086251896,
           "lookup": 1633086251900,
           "connect": 1633086252121,
           "secureConnect": 1633086252561,
           "upload": 1633086252561,
           "response": 1633086252835,
           "end": 1633086252835,
           "phases": {
             "wait": 0,
             "dns": 4,
             "tcp": 221,
             "tls": 440,
             "request": 0,
             "firstByte": 274,
             "download": 0,
             "total": 939
           }
         },
         "message": "Response code 404 (NOT FOUND)",
         "stack": "HTTPError: Response code 404 (NOT FOUND)\n    at Request.<anonymous> (/usr/src/app/node_modules/got/dist/source/as-promise/index.js:117:42)\n    at processTicksAndRejections (internal/process/task_queues.js:95:5)",
         "options": {
           "headers": {
             "user-agent": "RenovateBot/25.69.4 (https://github.com/renovatebot/renovate)",
             "authorization": "***********",
             "accept-encoding": "gzip, deflate, br"
           },
           "url": "https://api.bitbucket.org/2.0/repositories/sagar2089/renovate-config/src/HEAD/renovate.json",
           "username": "",
           "password": "",
           "method": "GET",
           "http2": false
         },
         "response": {
           "statusCode": 404,
           "statusMessage": "NOT FOUND",
           "body": "{\"type\":\"error\",\"error\":{\"message\":\"The requested repository either does not exist or you do not have access. If you believe this repository exists and you have access, make sure you're authenticated.\"}}",

Is there something that I need to define default.json or renovate.json in bitbucket. If so then can I get an example.

**Reply by ssharma2089** [reply]
@viceice  any update on the above error

**Reply by ssharma2089** [reply]
@viceice Could you please help me in configuration
